tests/tests.mk

```diff
@@ -0,0 +1,94 @@
+ISTIO_GO := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+export ISTIO_GO
+SHELL := /bin/bash -o pipefail
+
+VERSION ?= 1.6-dev
+
+# Base version of Istio image to use
+BASE_VERSION ?= 1.6-dev.0
+
+export GO111MODULE ?= on
+export GOPROXY ?= https://proxy.golang.org
+export GOSUMDB ?= sum.golang.org
+
+# If GOPATH is not set by the env, set it to a sane value
+GOPATH ?= $(shell cd ${ISTIO_GO}/../../..; pwd)
+export GOPATH
+
+# If GOPATH is made up of several paths, use the first one for our targets in this Makefile
+GO_TOP := $(shell echo ${GOPATH} | cut -d ':' -f1)
+export GO_TOP
+
+GO ?= go
+
+GOARCH_LOCAL := $(TARGET_ARCH)
+GOOS_LOCAL := $(TARGET_OS)
+
+HUB?=istio
+ifeq ($(HUB),)
+  $(error "HUB cannot be empty")
+endif
+
+# If tag not explicitly set in users' .istiorc.mk or command line, default to the git sha.
+TAG ?= $(shell git rev-parse --verify HEAD)
+ifeq ($(TAG),)
+  $(error "TAG cannot be empty")
+endif
+
+# Environment for tests, the directory containing istio and deps binaries.
+# Typically same as GOPATH/bin, so tests work seemlessly with IDEs.
+
+export ISTIO_BIN=$(GOBIN)
+# Using same package structure as pkg/
+
+export ISTIO_OUT:=$(TARGET_OUT)
+export ISTIO_OUT_LINUX:=$(TARGET_OUT_LINUX)
+
+# The following flags (in addition to ${V}) can be specified on the command-line, or the environment. This
+# is primarily used by the CI systems.
+_INTEGRATION_TEST_FLAGS ?= $(INTEGRATION_TEST_FLAGS)
+
+# $(CI) specifies that the test is running in a CI system. This enables CI specific logging.
+ifneq ($(CI),)
+	_INTEGRATION_TEST_FLAGS += --istio.test.ci
+	_INTEGRATION_TEST_FLAGS += --istio.test.pullpolicy=IfNotPresent
+endif
+
+ifeq ($(TEST_ENV),minikube)
+    _INTEGRATION_TEST_FLAGS += --istio.test.kube.minikube
+else ifeq ($(TEST_ENV),minikube-none)
+    _INTEGRATION_TEST_FLAGS += --istio.test.kube.minikube
+else ifeq ($(TEST_ENV),kind)
+    _INTEGRATION_TEST_FLAGS += --istio.test.kube.minikube
+endif
+
+ifneq ($(ARTIFACTS),)
+    _INTEGRATION_TEST_FLAGS += --istio.test.work_dir=$(ARTIFACTS)
+endif
+
+ifneq ($(HUB),)
+    _INTEGRATION_TEST_FLAGS += --istio.test.hub=$(HUB)
+endif
+
+ifneq ($(TAG),)
+    _INTEGRATION_TEST_FLAGS += --istio.test.tag=$(TAG)
+endif
+
+# $(INTEGRATION_TEST_KUBECONFIG) specifies the kube config file to be used. If not specified, then
+# ~/.kube/config is used.
+# TODO: This probably needs to be more intelligent and take environment variables into account.
+ifneq ($(KUBECONFIG),)
+	_INTEGRATION_TEST_FLAGS += --istio.test.kube.config=$(KUBECONFIG)
+else
+	_INTEGRATION_TEST_FLAGS += --istio.test.kube.config=~/.kube/config
+endif
+
+test.kube.presubmit: istioctl | $(JUNIT_REPORT)
```

Comment:
pretty sure `istioctl` won't work, right?

Reply:
hmm ... we need istioctl for the tests. Is there a way to download an istioctl binary?

Reply:
two choices. something like `go get istio.io/istio/istioctl/cmd/istioctl` or https://storage.googleapis.com/istio-build/dev/1.6-alpha.9fdbf4f32acb6edd967340930dd54a7cb2730f8e/istioctl-1.6-alpha.9fdbf4f32acb6edd967340930dd54a7cb2730f8e-linux.tar.gz